WIP: add tests for client_may_init and client_may_update

jwshea opened the pull request:
This pull request is for feedback, there's probably a better way than the naive cmp_deeply I've used...

Commit:
WIP: add tests

rjbs commented on t/basic.t:
```diff
@@ -2097,4 +2097,110 @@ subtest "string normalization" => sub {
+  ]);
+
+  cmp_deeply(
+    $jmap_tester->strip_json_types( $res->as_pairs ),
```
```
my $set = $res->single_sentence('biscuitsSet')->as_set;

my $foo_id =  $set->created_id('foo');
ok($foo_id, "we created foo");
jcmp_deeply(
  $set->create_errors->{bar},
  superhashof({
     propertyErrors => { qty => ignore() },
   }),
  }),
);
```

rjbs replied:
…and if it didn't work I might say:

```
unless ($tests_passed_or_whatever) {
  diag( explain( $res->as_stripped_pairs ));
}
```

Commit:
Code review changes

- use jcmp_deeply with "Set" sentence, instead of cmp_deeply on result
- added comments

jwshea commented:
I'd like to squash these before merging, just keeping separate to ease review; wasn't sure if github pull requests handle history rewrites as nicely as gitlab.

rjbs commented:
Yup, if you squash and force-push, it will track properly, at least in my conception of properly.

rjbs commented:
I have squashed and applied.
